### ISSUE TITLE
ReportManager rework

### ISSUE THREAD

**Issue by JMWren** [Everneth/EMI#89]
This class is in a similar state to the WhitelistAppService. The logic is hard to follow as it contains a lot of branches which could likely be reduced into a more human-readable state. The report commands are relatively unused so I am not even entirely sure it works properly, but I aim to push the use of these commands more and more. Just like the WhitelistAppService rework, this class should implement its own MessageReceived listener.

Perhaps the best course of action would be to simply remove reports to all the possible ministries and exclusively focus on reports to staff.

**Comment by TptMike** [comment]
All the logic pertaining to relaying messages to discord for those unsynced could be removed as those features are no longer needed. Since syncing is now required (and wasn't then), this should cut out at least half of the spaghet in the code.
